tests/net/mgmt/src/mgmt.c

```diff
@@ -423,4 +423,28 @@ ZTEST(mgmt_fn_test_suite, test_mgmt)
 		      "test_synchronous_event_listener failed");
 }
 
+static void net_mgmt_event_handler(struct net_mgmt_event_callback *cb,
+				    uint32_t mgmt_event, struct net_if *iface)
+{
+	ARG_UNUSED(cb);
+	ARG_UNUSED(iface);
+	ARG_UNUSED(mgmt_event);
+}
+
+ZTEST(mgmt_fn_test_suite, test_mgmt_duplicate_handler)
+{
+	struct net_mgmt_event_callback cb;
+
+	net_mgmt_init_event_callback(&cb, net_mgmt_event_handler, 0);
+	net_mgmt_add_event_callback(&cb);
+	net_mgmt_init_event_callback(&cb, net_mgmt_event_handler, 0);
+	net_mgmt_add_event_callback(&cb);
+
+	/* Wait for the callbacks to be processed */
+	k_sleep(K_MSEC(1000));
```

Comment:
I tried 100 and it didn't work (test passed even with the bug), 500 and 1000 work, but just to make sure we catch the bug in all platforms used the highest 1000.

Reply:
I would prefer that we remove the forced sleep and use a semaphore with a timeout here. The sem would be released in the callback, and if the semaphore timeouts (we need to check the return value), we can fail the test.
Waiting here always one second looks innocent, but these waits add up quickly when there are hundreds of tests that have sleeps in them, so lets try to avoid this here.